string filename
---
LineSegmentArray Path
ImpingementArray Impingement
